tests/unit/allowable-actions-mixin.spec.ts

```diff
@@ -0,0 +1,239 @@
+    expect(vm.isAllowed(undefined)).toBeNull()
+    expect(vm.isAllowed(null)).toBeNull()
+    expect(vm.isAllowed('')).toBeNull()
+  })
+
+  it('identifies whether Add Staff Comments is allowed', () => {
+    // initial state
+    expect(vm.isAllowed('addStaffComment')).toBe(false)
+
+    store.state.keycloakRoles = ['staff']
+    expect(vm.isAllowed('addStaffComment')).toBe(true)
+
+    // cleanup
+    store.state.keycloakRoles = []
+  })
+
+  it('identifies whether Dissolve Company is allowed', () => {
+    store.state.entityType = 'BC'
+
+    const tests = [
+      // no supported dissolution entities:
+      { value: [], expected: false },
+      // missing current dissolution entity:
+      { value: ['CP'], expected: false },
+      // with current dissolution entity:
+      { value: ['BC'], expected: true }
+    ]
+
+    for (let test of tests) {
+      getFeatureFlag.mockReturnValue(test.value)
+      expect(vm.isAllowed('dissolveCompany')).toBe(test.expected)
+    }
+
+    // cleanup
+    store.state.entityType = null
+  })
+
+  it('identifies whether Download Business Summary is allowed', () => {
+    getFeatureFlag.mockReturnValue(false)
+    expect(vm.isAllowed('downloadBusinessSummary')).toBe(false)
+
+    getFeatureFlag.mockReturnValue(true)
+    expect(vm.isAllowed('downloadBusinessSummary')).toBe(true)
+  })
+
+  it('identifies whether Edit Business Profile is allowed', () => {
+    const tests = [
+      // no conditions:
+      { isHistorical: true, businessId: null, expected: false },
+      // only first condition:
+      { isHistorical: false, businessId: null, expected: false },
+      // only second condition:
+      { isHistorical: true, businessId: 'CP1234567', expected: false },
+      // all conditions:
+      { isHistorical: false, businessId: 'CP1234567', expected: true }
+    ]
+
+    for (let test of tests) {
+      store.state.isHistorical = test.isHistorical
+      if (test.businessId) sessionStorage.setItem('BUSINESS_ID', test.businessId)
+      else sessionStorage.removeItem('BUSINESS_ID')
+      expect(vm.isAllowed('editBusinessProfile')).toBe(test.expected)
+    }
+  })
+
+  it('identifies whether File Address Change is allowed', () => {
+    const tests = [
+      // no conditions:
+      { isHistorical: true, hasBlocker: true, businessId: null, expected: false },
+      // only first condition:
+      { isHistorical: false, hasBlocker: true, businessId: null, expected: false },
+      // only second condition:
+      { isHistorical: true, hasBlocker: false, businessId: null, expected: false },
+      // only third condition:
+      { isHistorical: true, hasBlocker: true, businessId: 'CP1234567', expected: false },
+      // all conditions:
+      { isHistorical: false, hasBlocker: false, businessId: 'CP1234567', expected: true }
+    ]
+
+    for (let test of tests) {
+      store.state.isHistorical = test.isHistorical
+      jest.spyOn(vm, 'hasBlocker', 'get').mockReturnValue(test.hasBlocker)
```

Comment:
Make a note of this! It took me a while to figure out (Google was of little help).

This is how you mock a getter. ((Though in this case it's on `vm` object.)

Reply:
PS Trivia: `vm` stands for ViewModel in the [MVVM software architectural pattern](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel).